gpytorch/kernels/spectral_mixture_kernel.py

```diff
@@ -50,35 +142,58 @@ def initialize_from_data(self, train_x, train_y, **kwargs):
         train_x_sort = train_x.sort(1)[0]
         max_dist = train_x_sort[:, -1, :] - train_x_sort[:, 0, :]
         min_dist_sort = (train_x_sort[:, 1:, :] - train_x_sort[:, :-1, :]).squeeze(0)
-        min_dist = torch.zeros(1, self.n_dims)
-        for ind in range(self.n_dims):
+        min_dist = torch.zeros(1, self.ard_num_dims)
+        for ind in range(self.ard_num_dims):
             min_dist[:, ind] = min_dist_sort[(torch.nonzero(min_dist_sort[:, ind]))[0], ind]
 
         # Inverse of lengthscales should be drawn from truncated Gaussian | N(0, max_dist^2) |
         self.log_mixture_scales.data.normal_().mul_(max_dist).abs_().pow_(-1).log_()
         # Draw means from Unif(0, 0.5 / minimum distance between two points)
         self.log_mixture_means.data.uniform_().mul_(0.5).div_(min_dist).log_()
         # Mixture weights should be roughly the stdv of the y values divided by the number of mixtures
-        self.log_mixture_weights.data.fill_(train_y.std() / self.n_mixtures).log_()
+        self.log_mixture_weights.data.fill_(train_y.std() / self.num_mixtures).log_()
 
     def forward(self, x1, x2):
-        batch_size, n, n_dims = x1.size()
+        batch_size, n, num_dims = x1.size()
         _, m, _ = x2.size()
-        if not n_dims == self.n_dims:
-            raise RuntimeError("The number of dimensions doesn't match what was supplied!")
 
-        mixture_weights = self.log_mixture_weights.view(self.n_mixtures, 1, 1, 1).exp()
-        mixture_means = self.log_mixture_means.view(self.n_mixtures, 1, 1, 1, self.n_dims).exp()
-        mixture_scales = self.log_mixture_scales.view(self.n_mixtures, 1, 1, 1, self.n_dims).exp()
-        distance = (x1.unsqueeze(-2) - x2.unsqueeze(-3)).abs()  # distance = x^(i) - z^(i)
+        if not num_dims == self.ard_num_dims:
+            raise RuntimeError(
+                "The SpectralMixtureKernel expected the input to have %d dimensionality "
```

Comment:
Can we agree on the convention of `using.format()` for string formatting throughout? I find that much clearer.

Reply:
Sure I can fix that